modules/asset-manager-impl/src/main/java/org/opencastproject/assetmanager/impl/AssetManagerJobProducer.java

```diff
@@ -47,10 +49,10 @@
 import java.util.List;
 import java.util.function.Consumer;
 
-public class TieredStorageAssetManagerJobProducer extends AbstractJobProducer {
+public class AssetManagerJobProducer extends AbstractJobProducer {
```

Comment:
Currently, the asset-manager-workflowoperation pom.xml has _opencast-asset-manager-impl_ as a dependency.
It would be great to define an AssetManagerJobProducer interface in asset-manager-api with:
public interface AssetManagerJobProducer {
  public Job moveByIdAndDate(final String mpId, final Date start, final Date end, final String targetStorage);
  public Job moveByIdAndVersion(final Version version, final String mpId, final String targetStorage);
  public Job moveById(final String mpId, final String targetStorage);
  public Job moveByDate(final Date start, final Date end, final String targetStorage);
}
and then implement an AssetManagerJobProducerImpl in asset-manager-impl so that the asset-manager-workflowoperation pom.xml would depend on the api and not the impl.

Reply:
This does, kinda, make sense, since the WOHs depend on the specific implementation of `impl`.  That being said, I don't have any issue with this being done.

This could also be a chance to simplify the design a bit.  The original reason why there even *is* an `AMJPI` is because there was no way to extend two classes in a single subclass since Java doesn't do multiple inheritance.  I have not looked through the code here, but it's quite possible that the `AMJP` can be collapsed into the `AM` itself.

Reply:
I think what Greg is suggesting sounds like a good idea, but I don't have much more time to spend on this. But I bet there are a few low-hanging fruits to simplify left, so knock yourselves out! :D